msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -232,18 +301,35 @@ private static MessageBuffer newMessageBuffer(ByteBuffer bb)
             // We need to use reflection to create MessageBuffer instances in order to prevent TypeProfile generation for getInt method. TypeProfile will be
             // generated to resolve one of the method references when two or more classes overrides the method.
             return (MessageBuffer) mbBBConstructor.newInstance(bb);
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+        }
+        catch (InstantiationException e) {
+            // should never happen
+            throw new IllegalStateException(e);
+        }
+        catch (IllegalAccessException e) {
+            // should never happen unless security manager restricts this reflection
+            throw new IllegalStateException(e);
+        }
+        catch (InvocationTargetException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                // underlaying constructor may throw RuntimeException
+                throw (RuntimeException) e.getCause();
+            }
+            else if (e.getCause() instanceof Error) {
+                throw (Error) e.getCause();
+            }
+            // should never happen
+            throw new IllegalStateException(e.getCause());
         }
     }
 
     public static void releaseBuffer(MessageBuffer buffer)
     {
-        if (isUniversalBuffer || buffer.base instanceof byte[]) {
+        if (isUniversalBuffer || buffer.base != null) {
             // We have nothing to do. Wait until the garbage-collector collects this array object
```

Comment:
I think we should use `hasArray` instead of `buffer.base != null` here for future changes